addons/website/static/src/js/content/menu.js

```diff
@@ -120,6 +122,19 @@ const BaseAnimatedHeader = animations.Animation.extend({
             this.el.classList.remove('o_transitioning');
         }
     },
+    /**
+     * Scrolls to correctly display the section specified in the URL
+     *
+     * @private
+     */
+    _adjustUrlAutoScroll() {
+        // When the url contains #aRandomSection, prevent the navbar to overlap
+        // on the section, for this, we scroll as many px as the navbar height.
+        const wrapEl = document.querySelector('#wrapwrap');
+        if (!this.editableMode && wrapEl.scrollTop > 0) {
+            wrapEl.scrollBy(0, -this.el.offsetHeight);
```

Comment:
Actually there is already a place in the code where we detect if the page is loaded with a scroll > 0 ... It is in "_updateHeaderOnScroll". Maybe you have just to call the `_adjustUrlAutoScroll()` function there (in the` if (scroll > 0)` condition) and remove the scroll check no longer necessary in the function. Also it is preferable to use `$().getScrollingElement()[0]` rather than `document.querySelector('#wrapwrap');`. As `$().getScrollingElement()[0]` is already used in `_scrollHeightTooShort()`, you could put it in global variable.
And last thing, could you rewrite your commit message adding the steps to reproduce. It's not obvious that the bug is specific to certain header effect. Thanks ;)

Reply:
mhhh... I'm not a fan of calling `_adjustUrlAutoScroll` in `_updateHeaderOnScroll`. There is no point in calling this function every time the page is scrolled. Calling it in the start seems better to me because it will be done only once unlike `_updateHeaderOnScroll` which may be called dozens or even hundreds of times. For the rest you are right, I will adapt the code and the commit message :+1:

Reply:
> _updateHeaderOnScroll

Putting `_adjustUrlAutoScroll` in `_updateHeaderOnScroll` will be called only once if moved inside `if (scroll > 0)`..
Because of `if (!this.hasScrolled) {` ... This is the point of the `this.hasScrolled` variable.